env/node.fetcher/src/index.ts

```diff
@@ -8,26 +8,93 @@ import {
 } from '@pnpm/fetching-types'
 import { pickFetcher } from '@pnpm/pick-fetcher'
 import { createCafsStore } from '@pnpm/create-cafs-store'
+import { createHash } from '@pnpm/crypto.hash'
+import { type Cafs } from '@pnpm/cafs-types'
 import { createTarballFetcher } from '@pnpm/tarball-fetcher'
-import { type FetchFunction } from '@pnpm/fetcher-base'
+import { type NodeRuntimeFetcher, type FetchFunction, type FetchResult } from '@pnpm/fetcher-base'
+import { getNodeMirror, parseEnvSpecifier } from '@pnpm/node.resolver'
+import { addFilesFromDir } from '@pnpm/worker'
 import AdmZip from 'adm-zip'
 import renameOverwrite from 'rename-overwrite'
 import tempy from 'tempy'
 import { isNonGlibcLinux } from 'detect-libc'
 import ssri from 'ssri'
 import { getNodeArtifactAddress } from './getNodeArtifactAddress'
 
+export function createNodeRuntimeFetcher (ctx: {
+  fetch: FetchFromRegistry
+  rawConfig: Record<string, string>
+  offline?: boolean
+}): { nodeRuntime: NodeRuntimeFetcher } {
+  const fetchNodeRuntime: NodeRuntimeFetcher = async (cafs, resolution, opts) => {
+    if (!opts.pkg.version && !opts.pkg.id) {
+      throw new Error('Cannot fetch node.js without a version')
+    }
+    if (ctx.offline) {
+      throw new PnpmError('CANNOT_DOWNLOAD_NODE_OFFLINE', 'Cannot download Node.js because offline mode is enabled.')
+    }
+    // Sometimes the id comes in as runtime:<version> and sometimes as node@runtime:<version>.
+    // It would be nice to normalize this but unfortunately some parts of the code rely on IDs that start with the protocol.
+    const version = opts.pkg.version ?? opts.pkg.id.replace(/(?:node@)?runtime:/, '')
+    const { releaseChannel } = parseEnvSpecifier(version)
+
+    await validateSystemCompatibility()
+
+    const nodeMirrorBaseUrl = getNodeMirror(ctx.rawConfig, releaseChannel)
+    const artifactInfo = await getNodeArtifactInfo(ctx.fetch, version, {
```

Comment:
Slightly unrelated, but we have ["an official client for Node.js version information" from the PMWG](https://github.com/pkgjs/nv). I noticed below in files that are not changed here that it should support what you need and allow you to directly support the official Node.js lifecycle aliases *and* semver ranges.

Are you interested in exploring using that? If so I can look at it in detail and make sure our upcoming 1.0 release is fully compatible with your needs.

Reply:
It is a simple algorithm. I don't think we need a dependency for this.

Reply:
it might not stay that way if node changes their release schedule/lines, though.